### PR TITLE
Footer update

### PR DESCRIPTION
**Pull request by daRasmussen**
fixes issue 670 and updates footer param logic.

### CODE REVIEW

**Comment by jokd** [comment]
What about removing the urlText and just have text? If combined with a url then its a link, otherwise not. In your code you need to use urlText if you want image and text combined even if you have no url, which is not really obvious.

File: src/components/footer.js [hunk 1]
```diff
@@ -9,10 +9,18 @@ export default function Footer(options = {}) {
   return Component({
     render: function render() {
       let middleContent = '';
-      if (data.img) {
+      if (data.img && !data.url && !data.urlText) {
```
Comment (daRasmussen) [inline src/components/footer.js]: This statement is the same as original could be removed

Comment (jokd) [reply src/components/footer.js]: Agreed

File: src/components/footer.js [hunk 2]
```diff
@@ -9,10 +9,18 @@ export default function Footer(options = {}) {
...
+        middleContent = `<img src="${data.img}">`;
+      } else if (data.img && data.url && !data.urlText) {
+        middleContent = `<a href="${data.url}"><img src="${data.img}"></a>`;
+      } else if (data.img && !data.url && data.urlText) {
+        middleContent = `<img src="${data.img}">${data.urlText}`;
+      } else if (data.img && data.url && data.urlText) {
+        middleContent = `<img src="${data.img}"><a href="${data.url}">${data.urlText}</a>`;
```
Comment (jokd) [inline src/components/footer.js]: Should the image be a link as well?

File: src/components/footer.js [hunk 3]
```diff
@@ -9,10 +9,18 @@ export default function Footer(options = {}) {
...
+      } else if (data.img) {
         middleContent = `<img src="${data.img}">`;
       } else if (data.url && data.urlText) {
-        middleContent = `<a href="${data.url}">${data.urlText}</a>">`;
+        middleContent = `<a href="${data.url}">${data.urlText}</a>`;
       } else if (data.text) {
```
Comment (jokd) [inline src/components/footer.js]: How about replacing text with urlText (or better the other way around)?

File: src/components/footer.js (hunk 2 above)
Comment (daRasmussen) [reply src/components/footer.js]: sure, why not 👍

File: src/components/footer.js (hunk 3 above)
Comment (daRasmussen) [reply src/components/footer.js]: yeah for sure urlText, although I think this was text is from the original code so not sure if it was intended to be supported.

Comment (jokd) [reply src/components/footer.js]: However in case you have no url the urlText is a bit misleading

Comment (daRasmussen) [reply src/components/footer.js]: jupp looking at the docs, that is true. Maybe update docs with an statement that urlText is depended that the url for a link is set.

Comment (jokd) [reply src/components/footer.js]: yes, and that you cannot combine text with other elements. If you want image and text, you must use urlText.

Comment (daRasmussen) [reply src/components/footer.js]: Jupp! 

I will referense this conversion in an issue on the api-docs repo.

**Comment by tonnyandersson** [comment]
@daRasmussen I am unable to test this due to some local git issue preventing me from viewing external branches.

However, I agree with @jokd that the name _urlText_ is a bit misleading. I would like to suggest that we use three parameters - _img_, _url_ and _text_.

I can come up with six different scenarios, off the top of my head:

* img - shows an image only
* text - shows text only
* img + text - shows an image and text
* img + url - shows a clickable image
* text + url - shows clickable text
* img + text + url - shows an image with text, both clickable

I think I got them all, right?

This will probably screw things up with old config files, but I think it's worth it because it makes things more intuitive.

**Comment by jokd** [comment]
Closing this since #695 solves the issue

**Pull request closed without merging**